test/distributed/test_c10d.py

```diff
@@ -4127,6 +4129,49 @@ def test_broadcast_coalesced_gloo_cpu(self):
             self._test_broadcast_coalesced(process_group, device, root_rank)
 
 
+class TestDistributedNNFunctions(MultiProcessTestCase):
+    def setUp(self):
+        super(TestDistributedNNFunctions, self).setUp()
+        self._fork_processes()
+
+    def tearDown(self):
+        super(TestDistributedNNFunctions, self).tearDown()
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    @property
+    def op_timeout_sec(self):
+        return 1
+
+    @property
+    def world_size(self):
+        return 2
+
+    @requires_gloo()
+    @skip_if_not_multigpu
+    def test_broadcast(self):
+        store = c10d.FileStore(self.file_name, self.world_size)
+        # This is required because these functions calls directly to the .dist and needs
+        # the world to be initialized
+        dist.init_process_group(store=store, rank=self.rank, world_size=self.world_size, backend='gloo')
+        device = torch.device("cuda:%d" % self.rank)
+        x = torch.zeros(5, 5).to(device) + self.rank
+        x.requires_grad = True
+        y = torch.distributed.nn.broadcast(x, 1)
+        self.assertEqual(y, torch.ones(5, 5))
+        z = torch.nn.MSELoss()(x, y)
+        z.backward()
+
+        # def fn(x):
+        #     y = torch.distributed.nn.broadcast(x, 1)
+        #     y = torch.sin(y)
+        #     return torch.nn.MSELoss()(x, y)
+
+        # torch.autograd.gradcheck(fn, (x.clone().requires_grad_(),))
```

Comment:
Are we planning to add gradient checking and testing for the different distributed APIs we're planning to support?

Reply:
Ideally, some checks that verify the correctness of the grad must be added, but the calls itself in different APIs or backends are already tested, so I think that testing one is just enough.